apps/src/lib/node/ledger/protocol/transactions/ethereum_events/update.rs

```diff
@@ -0,0 +1,70 @@
+//! Helpers for writing to storage
+use borsh::{BorshDeserialize, BorshSerialize};
+use eyre::Result;
+use namada::ledger::storage::{DBIter, Storage, StorageHasher, DB};
+use namada::types::storage;
+use namada::types::token::Amount;
+
+/// Reads the `Amount` from key, applies update then writes it back
+pub(super) fn amount<D, H>(
+    store: &mut Storage<D, H>,
+    key: &storage::Key,
+    update: impl Fn(&mut Amount),
```

Comment:
```suggestion
    update: impl FnOnce(&mut Amount),
```
I think you can make this a `FnOnce`